packages/cli/src/modules/mcp/mcp.settings.controller.ts

```diff
@@ -36,4 +34,33 @@ export class McpSettingsController {
 		}
 		return { mcpAccessEnabled: enabled };
 	}
+
+	@GlobalScope('mcpApiKey:create')
+	@Get('/api-key')
+	async getApiKeyForMcpServer(req: AuthenticatedRequest) {
+		const apiKey = await this.mcpServerApiKeyService.findServerApiKeyForUser(req.user, true);
+
+		if (!apiKey) {
+			const newApiKey = await this.mcpServerApiKeyService.createMcpServerApiKey(req.user);
+			return newApiKey;
+		}
+
+		return apiKey;
+	}
+
+	@GlobalScope('mcpApiKey:rotate')
+	@Post('/api-key/rotate')
+	async rotateApiKeyForMcpServer(req: AuthenticatedRequest) {
+		const apiKey = await this.mcpServerApiKeyService.findServerApiKeyForUser(req.user);
+
+		if (!apiKey) {
+			throw new BadRequestError('No existing MCP server API key to rotate');
+		}
+
+		await this.mcpServerApiKeyService.deleteApiKeyForUser(req.user);
+
+		const newApiKey = await this.mcpServerApiKeyService.createMcpServerApiKey(req.user);
```

Comment:
This should happen in a single transaction. Otherwise the DB can be left in an inconsistent state